src/hooks/useUSDCPrice.ts

```diff
@@ -65,3 +70,54 @@ export function useUSDCValue(currencyAmount: CurrencyAmount<Currency> | undefine
     }
   }, [currencyAmount, price])
 }
+
+/**
+ *
+ * @param fiatValue string representation of a USD amount
+ * @returns CurrencyAmount where currency is stablecoin on active chain
+ */
+export function useStablecoinAmountFromFiatValue(fiatValue: string | null | undefined) {
+  const { chainId } = useActiveWeb3React()
+  const stablecoin = chainId ? STABLECOIN_AMOUNT_OUT[chainId]?.currency : undefined
+
+  if (fiatValue === null || fiatValue === undefined || !chainId || !stablecoin) {
+    return undefined
+  }
+
+  // trim for decimal precision when parsing
+  const parsedForDecimals = parseFloat(fiatValue).toFixed(stablecoin.decimals).toString()
+
+  try {
+    // parse USD string into CurrencyAmount based on stablecoin decimals
+    return tryParseAmount(parsedForDecimals, stablecoin)
+  } catch (error) {
+    return undefined
+  }
+}
+
+/**
+ * Get gas cost estimate for a dummy ETH / Stable v3 trade
+ * @returns Estimate in form of stablecoin amount or null
+ */
+export function useDefaultGasCostEstimate(): {
```

Comment:
feels like this function should be in a separate file `src/hooks/gas.ts` or something

Reply:
Yeah thought of this too - might move